File: apps/java/design-pattern/src/main/java/pattern/observer/TaxMan.java
package pattern.observer;

/**
 * 税務担当オブザーバー
 */
public class TaxMan implements Observer {

    private String lastNotification;

    @Override
    public void update(Employee employee) {
        lastNotification = employee.getName() + " に新しい税金の請求書を送付します";
    }

    public String getLastNotification() {
        return lastNotification;
    }
}

File: apps/php/design-pattern/src/Observer/TaxMan.php
<?php

declare(strict_types=1);

namespace DesignPattern\Observer;

class TaxMan implements Observer
{
    /** @var string[] */
    private array $log = [];

    public function update(Employee $employee): void
    {
        $this->log[] = sprintf(
            '%s に新しい税金の通知を送信しました',
            $employee->getName()
        );
    }

    /** @return string[] */
    public function getLog(): array
    {
        return $this->log;
    }
}
